src/pylorax/api/v0.py

```diff
@@ -2191,25 +2191,26 @@ def v0_upload_providers():
 @v0_api.route("/upload/provider/info/<provider_name>")
 @checkparams([("provider_name", "", "no provider given")])
 def v0_provider_info(provider_name):
-    """Return information about a provider, including its display name and
-    settings. Refer to the `resolve_provider` function."""
+    """Return information about a provider, including its display name,
+    expected settings, and saved profiles. Refer to the `resolve_provider`
+    function."""
     try:
         provider = resolve_provider(api.config["COMPOSER_CFG"]["upload"], provider_name)
     except RuntimeError as error:
         return jsonify(status=False, errors=[{"id": UPLOAD_ERROR, "msg": str(error)}])
     return jsonify(status=True, provider=provider)
 
-@v0_api.route("/upload/provider/update", defaults={"provider_name": ""}, methods=["POST"])
-@v0_api.route("/upload/provider/update/<provider_name>", methods=["POST"])
-@checkparams([("provider_name", "", "no provider given")])
-def v0_provider_update(provider_name):
+@v0_api.route("/upload/provider/save", defaults={"provider_name": ""}, methods=["POST"])
+@v0_api.route("/upload/provider/save/<provider_name>/<profile>", methods=["POST"])
```

Comment:
For our existing POST methods all information comes from the json we pass in, not from the URL. We should stay consistent with what we've done previously.

Reply:
I can move that stuff to the POST body, but it looks like we already pass variables in the URL in a few places? e.g. https://github.com/evan-goode/lorax/blob/a7cdb46518242fadfdfb659b2f8fd5603c291755/src/pylorax/api/v0.py#L526

Reply:
True, but those aren't also passing in a pile of settings. I think we should follow what's been done with creating new blueprints and sources and include it all in the json body.